compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/Util.java

```diff
@@ -61,4 +64,9 @@ static <T> T withClassLoader(ClassLoader classLoader, Callable<T> callable) thro
       Thread.currentThread().setContextClassLoader(appClassLoader);
     }
   }
+
+  static URI resolve(URI base, Class<? extends NessieApi> apiType) {
```

Comment:
`resolve` is a bit too generic - maybe `resolveNessieBaseUri` (with that you could use a static import for this method, too)